### PR TITLE
DifferentialActionModelContactFwdDynamics type agnostic

### PR DESCRIPTION
**Pull request by Diiso**
Hello,
This PR is related to issue #884. 
Changes:
- `DifferentialActionModelContactFwdDynamics` is type agnostic.
- benchmark/bipedal-timings.cpp was modified as it didn't have the `#include` floating base actuation model flag and was causing a compilation error due to change to `DifferentialActionModelContactFwdDynamics`.
- One unittest was added to test for `DifferentialActionModelContactFwdDynamics` with a `ActuationModelFull` instance (TalosArm is used for the test)

### CODE REVIEW

**Comment by hrp2-14** [comment]
Hi ! This project doesn't usually accept pull requests on master. If this wasn't intentionnal, you can change the base branch of this pull request to devel (No need to close it for that). Best, a bot.

File: bindings/python/crocoddyl/multibody/actions/contact-fwddyn.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ void exposeDifferentialActionContactFwdDynamics() {
       "is also a custom implementation in case of system with armatures. If you want to\n"
       "include the armature, you need to use setArmature(). On the other hand, the\n"
       "stack of cost functions are implemented in CostModelSum().",
-      bp::init<boost::shared_ptr<StateMultibody>, boost::shared_ptr<ActuationModelFloatingBase>,
+      bp::init<boost::shared_ptr<StateMultibody>, boost::shared_ptr<ActuationModelAbstract>,
                boost::shared_ptr<ContactModelMultiple>, boost::shared_ptr<CostModelSum>, bp::optional<double, bool> >(
           bp::args("self", "state", "actuation", "contacts", "costs", "inv_damping", "enable_force"),
           "Initialize the constrained forward-dynamics action model.\n\n"
```
Comment (cmastalli) [inline bindings/python/crocoddyl/multibody/actions/contact-fwddyn.cpp:27]: You need to update the documentation, do like this:
```cpp
 ":param actuation: actuation model\n"
```

**Review by jcarpent — COMMENTED** [review]
@Diiso Nice PR. Welcome to the club. 
Could you add the copyright of CTU and Inria on the top of the modified files?

File: benchmark/bipedal-timings.cpp [hunk 2]
```diff
@@ -26,6 +26,7 @@
 #include "crocoddyl/multibody/costs/state.hpp"
 #include "crocoddyl/multibody/costs/control.hpp"
 #include "crocoddyl/multibody/actuations/full.hpp"
+#include "crocoddyl/multibody/actuations/floating-base.hpp"
```
Comment (cmastalli) [inline benchmark/bipedal-timings.cpp:29]: It weird that this file doesn't compile for you.
If you pay attention, we need to include the `crocoddyl::ActuationModelFloatingBase`.

**Review by cmastalli — CHANGES_REQUESTED** [review]
Thanks for improving the API.
However, for me, it doesn't make sense that we need to remove the inclusion of the floating-base header (in bipedal-timings.cpp)


Indeed, this executable needs to include that header file.
Could you try to remove all  the cmake files and compile from scratch?

Finally, it seems the code doesn't have the right format. 
Here you can find instruction of how to do it automatically: https://github.com/loco-3d/crocoddyl/wiki/Developer-Guide

File: benchmark/bipedal-timings.cpp (hunk 2 above)
Comment (Diiso) [reply benchmark/bipedal-timings.cpp:29]: Hi @cmastalli, this file compiles fine if I include the floating base header in bibedal-timings.cpp. This `#include "crocoddyl/multibody/actuations/floating-base.hpp"` was absent from bipedal-timings.cpp previously.  
In more detail:
Previously, this include was in the `DifferentialActionModelContactFwdDynamics` file so when I modified the files include/crocoddyl/multibody/actions/contact-fwddyn, I replaced the floating base header by the abstract actuation model, causing the bidebal-timings.cpp to fail to compile. So I added the incude flag in bipedal-timings.cpp
Is this fine ?

**Comment by Diiso** [comment]
Thanks @jcarpent ! :) Copyrights added.

File: benchmark/bipedal-timings.cpp (hunk 2 above)
Comment (cmastalli) [reply benchmark/bipedal-timings.cpp:33]: Ok, I see.
Sorry I didn't realize it.

Thanks for the explanation

File: benchmark/bipedal-timings.cpp [hunk 3]
```diff
@@ -4,6 +4,10 @@
 // Copyright (C) 2018-2020, LAAS-CNRS
```
Comment (cmastalli) [inline benchmark/bipedal-timings.cpp:4]: The copyright is duplicate for CNRS, and it is not in the right place.
It should be
```cpp
///////////////////////////////////////////////////////////////////////////////
// BSD 3-Clause License
//
// Copyright (C) 2018-2020, LAAS-CNRS
// Copyright (C) 2020 CTU, INRIA
// Copyright note valid unless otherwise stated in individual files.
// All rights reserved.
///////////////////////////////////////////////////////////////////////////////
```

File: bindings/python/crocoddyl/multibody/actions/contact-fwddyn.cpp [hunk 4]
```diff
@@ -4,6 +4,10 @@
 // Copyright (C) 2018-2020, LAAS-CNRS, University of Edinburgh
 // Copyright note valid unless otherwise stated in individual files.
 // All rights reserved.
+//
```
Comment (cmastalli) [inline bindings/python/crocoddyl/multibody/actions/contact-fwddyn.cpp:7]: Replicate similar pattern here, and in the other files. Thanks :)

File: benchmark/bipedal-timings.cpp [hunk 5]
```diff
@@ -4,6 +4,10 @@
...
 // Copyright note valid unless otherwise stated in individual files.
 // All rights reserved.
+//
+// Copyright (c) 2020 CNRS, INRIA
```
Comment (jcarpent) [inline benchmark/bipedal-timings.cpp:8]: You can remove CNRS, and put CTU and INRIA on the same row.

File: benchmark/bipedal-timings.cpp (hunk 3 above)
Comment (Diiso) [reply benchmark/bipedal-timings.cpp:4]: Thank you for giving me the format ! I didn't know what it was.

**Review by cmastalli — APPROVED** [review]
Thanks @Diiso

File: unittest/factory/diff_action.hpp [hunk 6]
```diff
@@ -1,7 +1,8 @@
 ///////////////////////////////////////////////////////////////////////////////
 // BSD 3-Clause License
 //
-// Copyright (C) 2018-2020, University of Edinburgh
+// Copyright (C) 2018-2020, LAAS-CNRS
```
Comment (jcarpent) [inline unittest/factory/diff_action.hpp:4]: Why did you change this?

Comment (Diiso) [reply unittest/factory/diff_action.hpp:4]: Oops my bad, I did a copy paste and I didn't notice `University of Edinburgh`. Sorry about that, I will change that.

**Review by proyan — APPROVED** [review]
